src/sentry/profiles/task.py

```diff
@@ -346,6 +362,52 @@ def _process_symbolicator_results_for_rust(profile: Profile, stacktraces: List[A
             original["frames"] = symbolicated["frames"]
 
 
+"""
+This function returns a map {index: [indexes]} that will let us replace a specific
+frame index with (potentially) a list of frames indices that originated from that frame.
+
+The reason for this is that the frame from the SDK exists "physically",
+and symbolicator then synthesizes other frames for calls that have been inlined
+into the physical frame.
+
+Example:
+
+`
+fn a() {
+b()
+}
+fb b() {
+fn c_inlined() {}
+c_inlined()
+}
+`
+
+this would yield the following from the SDK:
+b -> a
+
+after symbolication you would have:
+c_inlined -> b -> a
+
+The sorting order is callee to caller (child to parent)
+"""
+
+
+def get_frame_index_map(frames: List[dict[str, Any]]) -> dict[int, List[int]]:
+    counter = 0
+    index_map = {}
+    for i, frame in enumerate(frames):
+        if frame["status"] != "symbolicated":
```

Comment:
As we were discussing in a call, using `frame["original_index"]` instead of the `counter` might be best solution.

The "status" can be lots of different things like "missing symbol", etc. And `sym_addr` is not being used for example for .NET now that we support that.

The `original_index` is exactly the thing you want to be able to refer back to the original frame that you sent to symbolicator.